src/Evolution/Systems/Cce/AnalyticSolutions/TeukolskyWave.cpp

```diff
@@ -0,0 +1,326 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/Cce/AnalyticSolutions/TeukolskyWave.hpp"
+
+#include <complex>
+#include <cstddef>
+#include <memory>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/SpinWeighted.hpp"
+#include "DataStructures/Tags.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/WorldtubeData.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCollocation.hpp"
+#include "NumericalAlgorithms/Spectral/SwshInterpolation.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Cce::Solutions {
+
+/// \cond
+TeukolskyWave::TeukolskyWave(const double extraction_radius,
+                             const double amplitude,
+                             const double duration) noexcept
+    : SphericalMetricData{extraction_radius},
+      amplitude_{amplitude},
+      duration_{duration} {}
+
+std::unique_ptr<WorldtubeData> TeukolskyWave::get_clone() const noexcept {
+  return std::make_unique<TeukolskyWave>(*this);
+}
+
+double TeukolskyWave::pulse_profile_coefficient_a(const double time) const
+    noexcept {
+  const double retarded_time = time - extraction_radius_;
+  return 3.0 *
+         (3.0 * pow<4>(duration_) +
+          4.0 * square(extraction_radius_) * square(retarded_time) -
+          2.0 * square(duration_) * extraction_radius_ *
+              (extraction_radius_ + 3.0 * retarded_time)) *
+         amplitude_ * exp(-square(retarded_time) / square(duration_)) /
+         (pow<4>(duration_) * pow<5>(extraction_radius_));
+}
+
+double TeukolskyWave::pulse_profile_coefficient_b(const double time) const
+    noexcept {
+  const double retarded_time = time - extraction_radius_;
+  return 2.0 *
+         (-3.0 * pow<6>(duration_) +
```

Comment:
this does not agree with the documentation, the leading term should be 1/R^5 according to the doc

Reply:
can you make the code and documentation factored the same so it is easier to visually compare?